docs/migration.rst

```diff
@@ -0,0 +1,338 @@
+convert the ``.coords`` attribute instead::
+
+    >>> line.coords
+    <shapely.coords.CoordinateSequence at 0x7f2e09e88d60>
+    >>> np.array(line.coords)
+    array([[0., 0.],
+           [1., 1.],
+           [2., 2.]])
+
+The ``array_interface()`` method and ``ctypes`` attribute will be removed in
+Shapely 2.0, but since Shapely will start requiring NumPy as a dependency,
+you can use NumPy or its array interface directly.
+
+Creating NumPy arrays of geometry objects
+-----------------------------------------
+
+Shapely geometry objects can be stored in NumPy arrays using the ``object``
+dtype. In general, one could create such an array from a list of geometries
+as follows::
+
+    >>> from shapely.geometry import Point
+    >>> arr = np.array([Point(0, 0), Point(1, 1), Point(2, 2)])
+    >>> arr
+    array([<shapely.geometry.point.Point object at 0x7fb798407cd0>,
+           <shapely.geometry.point.Point object at 0x7fb7982831c0>,
+           <shapely.geometry.point.Point object at 0x7fb798283b80>],
+          dtype=object)
+
+The above works for point geometries, but because in Shapely 1.x, some
+geometry types are sequence-like (see above), NumPy can try to "unpack" them
+when creating an array. Therefore, for more robust creation of a NumPy array
+from a list of geometries, it's generally recommended to this in a two-step
+way (first creating an empty array and then filling it)::
+
+    geoms = [Point(0, 0), Point(1, 1), Point(2, 2)]
+    arr = np.empty(len(geoms), dtype="object")
+    arr[:] = geoms
+
+This code snippet results in the same array as the example above, and works
+for all geometry types and Shapely/NumPy versions. 
+
+However, starting with Shapely 1.8, the above code will show deprecation
+warnings that cannot be avoided (depending on the geometry type, NumPy tries
+to access the array interface of the objects or check if an object is
+iterable or has a length, and those operations are all deprecated now. The
+end result is still correct, but the warnings appear nonetheless).
+Specifically in this case, it is fine to ignore those warnings (and the only
+way to make them go away)::
+
+    import warnings
+    from shapely.errors import ShapelyDeprecationWarning
+
+    geoms = [Point(0, 0), Point(1, 1), Point(2, 2)]
+    arr = np.empty(len(geoms), dtype="object")
+
+    with warnings.catch_warnings():
+        warnings.filterwarnings("ignore", category=ShapelyDeprecationWarning)
+        arr[:] = geoms
+
+In Shapely 2.0, the geometry objects will no longer be sequence like and
+those deprecation warnings will be removed (and thus the ``filterwarnings``
+will no longer be necessary), and creation of NumPy arrays will generally be
+more robust.
+
+If you maintain code that depends on Shapely, and you want to have it work
+with multiple versions of Shapely, the above code snippet provides a context
+manager that can be copied into your project::
+
+    import contextlib
+    import shapely
```

Comment:
If including this full code snippet might make the section a bit long to read, we could alternatively also link to the github issue's comment from where I copied this: https://github.com/Toblerity/Shapely/issues/1096#issuecomment-789998514 (which might also make it easier to comment / ask questions about it on the issue)